### ISSUE TITLE
If sync-semver-tags is disabled the release tag is not verified

### ISSUE THREAD

**Issue by davideroffo** [nearform/optic-release-automation-action#216]
We noticed that if the `sync-semver-tags` input parameter is disabled the release tag (when created by GitHub itself) is not verified.

### To be verified:
Probably we should apply the same logic we are applying in case of `sync-semver-tags` enabled for the specific release tag.

Current code:

```javascript
try {
    const syncVersions = /true/i.test(inputs['sync-semver-tags'])

    if (syncVersions && !isPreRelease) {
      const { major, minor, patch } = semver.parse(version)

      await tagVersionInGit(`v${major}`)
      await tagVersionInGit(`v${major}.${minor}`)
      await tagVersionInGit(`v${major}.${minor}.${patch}`)
    }
  } catch (err) {
    core.setFailed(`Unable to update the semver tags ${err.message}`)
  }
```

Expected code:

```javascript
try {
    const { major, minor, patch } = semver.parse(version)
    await tagVersionInGit(`v${major}.${minor}.${patch}`)

    const syncVersions = /true/i.test(inputs['sync-semver-tags'])

    if (syncVersions) {
      await tagVersionInGit(`v${major}`)
      await tagVersionInGit(`v${major}.${minor}`)
    }
  } catch (err) {
    core.setFailed(`Unable to update the semver tags ${err.message}`)
  }
```

### Example with `sync-semver-tags` disabled

<img width="1128" alt="image" src="https://user-images.githubusercontent.com/6604111/208652199-1b2c7320-1753-4fc7-9e26-12b51e092389.png">

### Example with `sync-semver-tags` enabled

<img width="1119" alt="image" src="https://user-images.githubusercontent.com/6604111/208652280-46153a1d-d07b-4d98-8bf1-1ba80bcb16d8.png">

**Comment by simoneb** [comment]
Let's double check that the assumptions made in this issue are actually right, because we have seen fairly inconsistent behaviors across various repositories.

**Comment by davideroffo** [comment]
Unfortunately, the assumption I made doesn't seem correct, because the `verified` tag is randomly getting assigned even if the `sync-semver-tags` option is enabled.

I was trying to verify the commits and the tags with the `-S` parameter with the `git` command but it requires the configuration and implementation of GPG to sign the commit (or the tag).

I then imported a new action in the [linked PR](https://github.com/nearform/optic-release-automation-action/pull/222) in order to import GPG and configure it but I'm stuck on making it works fine.

I agreed with @simoneb to pause the implementation of this fix because is not a real "issue" or "bug": it is not causing any side effect, and focusing on other priorities.
